lib/force.py

```diff
@@ -280,121 +281,119 @@
     "each": ["fakeroot"] },
 }
 
+# Default value of --force-cmd.
+#
+# NOTE: apt(8) tells people not to use it in scripts, but they do it anyway.
+FORCE_CMD_DEFAULT = { "apt":     ["-o", "APT::Sandbox::User=root"],
+                      "apt-get": ["-o", "APT::Sandbox::User=root"] }
+
 
 ## Functions ###
 
-def detect(image, force, no_force_detect):
-   f = None
-   if (no_force_detect):
-      ch.VERBOSE("not detecting --force config, per --no-force-detect")
+def new(image_path, force_mode, force_cmds):
+   """Return a new forcer object appropriate for image at image_path in mode
+      force_mode. If no such object can be found, exit with error."""
+   if (force_mode is None):
+      return Noop()
+   elif (force_mode == "fakeroot"):
+      return Fakeroot(image_path)
+   elif (force_mode == "seccomp"):
+      return Seccomp(force_cmds)
    else:
-      # Try to find a real fakeroot config.
-      for (tag, cfg) in DEFAULT_CONFIGS.items():
-         try:
-            f = Fakeroot(image, tag, cfg, force)
-            break
-         except Config_Aint_Matched:
-            pass
-      # Report findings.
-      if (f is None):
-         msg = "--force not available (no suitable config found)"
-         if (force):
-            ch.WARNING(msg)
-         else:
-            ch.VERBOSE(msg)
-      else:
-         if (force):
-            adj = "will use"
-         else:
-            adj = "available"
-         ch.INFO("%s --force: %s: %s" % (adj, f.tag, f.name))
-   # Wrap up
-   if (f is None):
-      f = Fakeroot_Noop()
-   return f
+      assert False, "unreachable code reached"
+
+def force_cmd_parse(text):
+   # 1. Split on “,” preceded by even number of backslashes.
+   #
+   # FIXME: Said backslashes are removed in the split, so you can’t have a
+   # component with trailing backslashes. That seems rare so I’m not fixing
+   # for now.
+   args = re.split(r"(?<!\\)(?:\\\\)*,", text)
```

Comment:
I'm not sure I'm understanding the `FIXME` correctly, but would something like this help?
```python
>>> x = re.compile(r"((?<!\\)(?:\\\\)*,)")
>>> m = x.split(r"echo,I,can't,run,my,command,apt\,fix\,all-the-things\,--please,on,my,C:\\,drive")
>>> print(m)
['echo', ',', 'I', ',', "can't", ',', 'run', ',', 'my', ',', 'command', ',', 'apt\\,fix\\,all-the-things\\,--please', ',', 'on', ',', 'my', ',', 'C:', '\\\\,', 'drive']

>>> print(m[0::2])
['echo', 'I', "can't", 'run', 'my', 'command', 'apt\\,fix\\,all-the-things\\,--please', 'on', 'my', 'C:', 'drive']
>>> print(m[1::2])
[',', ',', ',', ',', ',', ',', ',', ',', ',', '\\\\,']
```
Parenthesizing the whole RE like that saves each delimiter string in between the fields themselves, as you can see in `m` above, and it's relatively painless to pull out/separate the two individual lists.

Reply:
Yeah that's not quite right (I think the desired output is `['echo', 'I', "can't", 'run', 'my', 'command', 'apt,fix,all-the-things,--please', 'on', 'my', 'C:\drive']`) but I'm still having a hard time grokking exactly what's up. The problem had to do with strings that terminate in a backslash.